ui/components/NodeInputSubmit.tsx

```diff
@@ -43,9 +50,50 @@ export const NodeInputSubmit: FC<NodeInputProps> = ({
     (node.meta.label as unknown as { hasTotpLink: boolean })?.hasTotpLink ??
     false;
   const showBackLink = node.meta?.label?.text === "Reset password";
+  const showBackupCodesIntro =
```

Comment:
In general it feels weird to have backup codes specific logic here as `NodeInputSubmit` seems to be a general purpose component. I lack a bit of historical context for this project, is it not possible to handle the logic for modal, checkbox and intro at a parent level and pass down the components for rendering?

Reply:
+1

Reply:
Yes, good idea to keep the specific logic in a higher level and not leak it into this generic component. Here we could just have a "render before" and "render after" component, that is put around the button to keep the interface and purpose clean and generic.